local_planner/include/local_planner/local_planner_visualization.h

```diff
@@ -0,0 +1,164 @@
+#ifndef LOCAL_PLANNER_VISUALIZATION_H
+#define LOCAL_PLANNER_VISUALIZATION_H
+
+#include "local_planner/local_planner.h"
+
+#include <pcl/point_cloud.h>
+#include <pcl_ros/point_cloud.h>
+#include <ros/ros.h>
+#include <Eigen/Dense>
+#include <vector>
+
+namespace avoidance {
+
+class LocalPlannerVisualization {
+ public:
+  /**
+  * @brief      initializes all subscribers used for local planner visualization
+  **/
+  void initializeSubscribers(ros::NodeHandle& nh);
+
+  /**
+  * @brief       Main function which calls functions to visualize all planner
+  *output ready at the end of one planner iteration
+  * @params[in] planner, reference to the planner
+  * @params[in] newest_waypoint_position, last caluclated waypoint (smoothed)
+  * @params[in] newest_adapted_waypoint_position, last caluclated waypoint
+  *(non-smoothed)
+  **/
+  void visualizePlannerData(
+      LocalPlanner& planner,
+      const geometry_msgs::Point& newest_waypoint_position,
+      const geometry_msgs::Point& newest_adapted_waypoint_position,
+      const geometry_msgs::PoseStamped& newest_pose);
+
+  /**
+  * @brief       Visualization of the calculated search tree and the best path
+  *chosen
+  * @params[in]  tree, the complete calculated search tree
+  * @params[in]  closed_set, the closed set (all expanded nodes)
+  * @params[in]  path_node_positions, the positions of all nodes belonging to
+  *the chosen best path
+  **/
+  void publishTree(std::vector<TreeNode>& tree,
+                   const std::vector<int> closed_set,
```

Comment:
There's no point in passing a const value.
Either const& or value (here and all following functions too)

Reply:
Yep, I think const& was intended here, and a few other places too.